[WIP] Val 2i - Deuterium retention in neutron-irradiated single-crystal tungsten

cticenhour opened the pull request:
Masa's Paper: https://www.sciencedirect.com/science/article/pii/S0920379618303880

Closes #393

cticenhour committed:
Add temperature series plotting script for ISFNT case

cticenhour committed:
Add initial val-2i input file

Confirmed that temperature and diffusivity profiles over time look reasonable, and there is a reasonably high trapped population, but zero mobile concentration. Convergence fails over only a couple of timesteps.

cticenhour committed:
Refine simple mesh, turn on timestep limiting guidelines in adaptive timestepper, and lower trap_per_free parameter

cticenhour committed:
Add source aux for troubleshooting

cticenhour committed:
Allow easier adjustment of font sizes for labels and axes for temperature ploting script

cticenhour committed:
Add lower bounds for concentration and trapped_1 variables

cticenhour committed:
Add reflection coefficient impact to source term to be more consistent with paper

cticenhour committed:
Parameter adjustments to be more consistent with paper usage of trap concentration, and other parameter adjustments for clarity

cticenhour committed:
Switch to CartesianMeshGenerator to refine in source region a bit better

cticenhour committed:
Adjust trap_per_free and refine in implantation region to get to 8148 s

cticenhour committed:
Adjust tolerances to get case running and add initial plotting script

cticenhour committed:
Make adjustments to better match simulation and presentation of results with the source publication

cticenhour committed:
Adjustments to clean up commented out items or otherwise unnecessary settings in input file, that did not impact results.

cticenhour committed:
Add TMAP4 run data to plot comparison script

cticenhour committed:
Add Masa's experimental data for comparison

Use random sampling to reduce the density of the scatter plot of the experimental data - similar to what Masa's doing in the paper. Cannot tell what approach he is using the reduce the plot density.

As of this change, 10% of the data will be sampled and shown on the plot using the random sampler.

cticenhour committed:
Improve readability of plot by adjusting grid and alpha (transparency) parameters

simopier commented on test/tests/val-2g/comparison_val-2g.py:
```diff
@@ -0,0 +1,111 @@
+# tmap_flux_for_rmspe = numerical_solution_on_experiment_input(experiment_time, simulation_time_TMAP7, simulation_flux_left_TMAP7/2 + flux_environment)
+# RMSE = np.sqrt(np.mean((tmap_flux_for_rmspe-experiment_flux)**2) )
+# RMSPE = RMSE*100/np.mean(experiment_flux)
+# ax.text(6000.0,0.85e18, 'RMSPE = %.2f '%RMSPE+'%',fontweight='bold')
```
Why is this commented out? Maybe it's just because this is still in WIP stage, but calculating the RMSPE between TMAP8's predictions and the experimental data is a good thing to add.

cticenhour replied:
RMSPE values (for TMAP4 and TMAP8 simulation results against experimental data.....thought it would be an interesting point of comparison) were added in 8506bf3. Though, there are some quirks that needs to be ironed out. TMAP4 RMSPE looks reasonable, but TMAP8 RMSPE is off the charts (+15000%). Looks like there is an error or the data needs to be "cleaned" - like, we need to compensate for the environment or something. Need to explore further.

cticenhour replied:
Looks like the full data set has quite a bit of noise that was truncated out in the data plot: 

<img width="1902" height="1511" alt="val-2i_comparison_full" src="https://github.com/user-attachments/assets/3e927e46-59d8-40a9-a083-dcc81c034240" />

Do you have a good way off the top of your head to remove this "environment flux"? I see that an environment flux was calculated and used in the val-2d comparison script, but I am not sure if we can do the same here. Probably need to review the context in the documentation, first.

simopier replied:
Are you talking about the values at 10^5?  this sounds like basically 0, and it probably corresponds to the minimum detection limit. I don't think it's meaningful in any way. Masa might be able to confirm or deny that.

simopier commented on test/tests/val-2g/val-2g_temperature.py:
```diff
@@ -0,0 +1,113 @@
+                                         temperature_room, t, hold_time[i]))
+        elif t < TDS_ramp_end:
+            # Ramp phase
+            temp.append(temperature_ramp_rate * (t - TDS_start) + temperature_min)
```
```suggestion
            temperature.append(temperature_ramp_rate * (t - TDS_start) + temperature_min)
```

simopier commented on test/tests/val-2g/val-2g_temperature.py:
```diff
@@ -0,0 +1,113 @@
+            # Final hold phase
+            temp.append(final_temperature)
+
+    cooling_data.append(temp)
```
```suggestion
    cooling_data.append(temperature)
```

simopier commented on test/tests/val-2g/val-2g.i:
```diff
@@ -0,0 +1,321 @@
+  []
+[]
+
+[BCs]
+  [left]
+    type = ADDirichletBC
+    variable = concentration
+    boundary = left
+    value = 0
+  []
+  [right]
+    type = ADDirichletBC
+    variable = concentration
+    boundary = right
+    value = 0
+  []
+[]
```
This is a reasonable assumption, but I want to highlight that this assumption was shown to significantly affect the results compared to accounting for finite recombination kinetics at surfaces in val-2f (https://mooseframework.inl.gov/TMAP8/verification_and_validation/val-2f.html). I'm not saying the same thing should be done here, but maybe it could be mentionned in the documentation of this validation case with a link to val-2f.

cticenhour replied:
I made an issue to improve the overall case (in the next PR) in #395. Will add a blurb about this and link a pending commit.

simopier commented on test/tests/val-2g/val-2g_temperature.py:
I understand plotting the temperature history this way, but I think there is also value in plotting the tmeperature history based on the temperature from the csv outputs of the TMAP8 simulations. The advantage of doing so is that it will be updated along with the TMAP8 input files (if it changes or is adapted for a different history), and would clearly highlight any potential issues with the implementation of the temperature history in the input file. It also prevents from having the temperature history implemented in two different places (this scipt and the input file). Let me know what you think.

cticenhour replied:
Addressed across a couple of commits, but finished off in b2d204c.

simopier reviewed (changes requested):
Thank you @cticenhour! I'm excited for this case to be added!

I left comments and suggestions even though this is still tagged as [WIP] because I was too curious to ignore it. It looks like the main thing missing is the documentation, correct?

cticenhour committed:
Apply suggestions from code review

Co-authored-by: Pierre-Clement Simon <pierreclement.simon@gmail.com>

cticenhour committed:
Add units to experimental and TMAP4 data for val-2i

cticenhour committed:
Add header information to val-2i input file

cticenhour commented on test/tests/val-2g/comparison_val-2g.py:
```diff
@@ -0,0 +1,111 @@
+# Setup mask using a random sampling method to reduce density of scatter plot points in experimental
+# data (there is a lot of "noise").
+# These preserve 10% of the data points. (adjusted within the 'size' parameter).
+mask = np.random.choice(len(experiment_flux_673), size=int(len(experiment_flux_673) * 0.1), replace=False)
```
Sent you a message just now on Teams - interested in hearing what you think! Left some context on "why" there as well.

cticenhour replied:
Mask has been removed and alpha has been set to compensate for the number of points against the simulation line results.

cticenhour commented on test/tests/val-2g/tmap4_data.csv:
Added in adad02e.

cticenhour commented on test/tests/val-2g/val-2g.i:
```diff
@@ -0,0 +1,321 @@
+# General parameters
```
Added in 3508a25.

cticenhour committed:
Adjust temperature plotting script to include temperature profile directly from the simulation.

Add temperature function to postprocessor output for use in scripts.

Start output of simulation data from timestep 0 to get the full temperature series.

cticenhour committed:
Add initial val-2i tests

Refs #393

cticenhour committed:
Add light test for val-2i and mark full tests as heavy

cticenhour committed:
Add val-2i documentation

cticenhour committed:
Fixup typo in comparison script when used with documentation

cticenhour committed:
Change name of temperature history generation file and output file

cticenhour committed:
Address review comments

- Fix typo in temperature history script. Now always uses gold files from TMAP8 simulation.
- Add units comments to postprocessor outputs

cticenhour commented on test/tests/val-2i/val-2i.i:
```diff
@@ -0,0 +1,321 @@
+[]
+
+[Postprocessors]
+  [flux_surface_left]
```
Addressed in b2d204c.

cticenhour commented on test/tests/val-2i/val-2i.i:
```diff
@@ -0,0 +1,321 @@
+    boundary = 'right'
+    outputs = none
+  []
+  [scaled_flux_surface_right]
```
Addressed in b2d204c.

cticenhour commented on test/tests/val-2i/val-2i.i:
```diff
@@ -0,0 +1,321 @@
+    execute_on = 'initial nonlinear linear timestep_end'
+    outputs = 'console csv exodus'
+  []
+  [flux_surface_right]
```
Addressed in b2d204c.

cticenhour commented on test/tests/val-2i/val-2i.i:
```diff
@@ -0,0 +1,321 @@
+    boundary = 'left'
+    outputs = none
+  []
+  [scaled_flux_surface_left]
```
Addressed in b2d204c.

cticenhour committed:
Remove data masking and use alphas instead

cticenhour committed:
Add RMSPE for both TMAP4 results and TMAP8 results against experimental data

cticenhour committed:
Add to caption about RMSPE values

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/3668219/), step Python: black format on 73ff72b wanted to post the following:

## Python black formatting

Your code requires style changes.

A patch was generated and copied [here](https://mooseframework.inl.gov/tmap8/docs/PRs/325/black/black.patch).

You can directly apply the patch by running the following at the top level of your repository:

```
curl -s https://mooseframework.inl.gov/tmap8/docs/PRs/325/black/black.patch | git apply -v
```

Alternatively, you can run the following at the top level of your repository:

```
black --config pyproject.toml --workers 1 .
```

cticenhour committed:
Perform black formatting on val-2i python files

moosebuild commented:
Job [Build test summary](https://civet.inl.gov/job/3668562/), step Build test summary on b41cfdf wanted to post the following:

## Test summary

Compared against c837769 in job [civet.inl.gov/job/3665797](https://civet.inl.gov/job/3665797).

### Added tests

| Test                            | Time (s)   | Memory (MB)   |
|---------------------------------|------------|---------------|
| `test:val-2i.val-2i_673K_light` | 4.68       | 204.66        |
| `test:val-2i.val-2i_comparison` | 2.43       | 149.25        |
| `test:val-2i.val-2i_673K`       | SKIP       | 0.00          |
| `test:val-2i.val-2i_873K`       | SKIP       | 0.00          |
| `test:val-2i.val-2i_973K`       | SKIP       | 0.00          |

### Run time changes

| Test                                     | Base (s)   | Head (s)   | +/-     | Base (MB)   | Head (MB)   |
|------------------------------------------|------------|------------|---------|-------------|-------------|
| `test:ver-1fb.thermal_transient_physics` | 4.58       | 7.14       | +55.91% | 53.46       | 59.70       |
| `test:ver-1fb.ver-1fb`                   | 2.44       | 3.69       | +51.01% | 43.90       | 44.89       |

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/3672022/), step Sync to remote on 8a04362 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/tmap8/docs/PRs/325/site)

This comment will be updated on new commits.

cticenhour committed:
Simplify outputs setup

Some of this was needed during debug, but is no longer.

This change also makes it easier to turn off Exodus output for testing.

cticenhour committed:
Switch to MUMPS and reduce bounds min to zero

The MUMPS change will hopefully help with robustness in parallel / distributed mesh

cticenhour committed:
Differentiate requirements in val-2i tests

simopier commented on doc/content/bib/tmap8.bib:
```diff
@@ -610,3 +610,63 @@ @article{yu2024hydrogen
+
+@article{Katoh2017phenix,
+  author = {Yutai Katoh and Daniel Clark and Yoshio Ueda and Yuji Hatano and Minami Yoda and Adrian S. Sabau and Takehiko Yokomine and Lauren M. Garrison and J. Wilna Geringer and Akira Hasegawa and Tatsuya Hinoki and Masashi Shimada and Dean Buchenauer and Yasuhisa Oya and Takeo Muroga},
+  title = {Progress in the U.S./Japan PHENIX Project for the Technological Assessment of Plasma Facing Components for DEMO Reactors},
```
```suggestion
  title = {{Progress in the U.S./Japan PHENIX Project for the Technological Assessment of Plasma Facing Components for DEMO Reactors}},
```

simopier commented on doc/content/bib/tmap8.bib:
```diff
@@ -610,3 +610,63 @@ @article{yu2024hydrogen
+
+@article{Shimada2017phenix,
+  author = {Masashi Shimada and Yasuhisa Oya and Dean A. Buchenauer and Yuji Hatano},
+  title = {Hydrogen Isotope Retention and Permeation in Neutron-Irradiated Tungsten and Tungsten Alloys Under PHENIX Collaboration},
```
```suggestion
  title = {Hydrogen Isotope Retention and Permeation in Neutron-Irradiated Tungsten and Tungsten Alloys Under {PHENIX} Collaboration},
```

simopier commented on doc/content/verification_and_validation/val-2i.md:
```diff
@@ -0,0 +1,177 @@
+## Case Description
+
+!style halign=left
+This case reproduces, in updated form, the analysis published in [!cite](Shimada2018), where a modified form of the TMAP4 code (updated to include multiple trapping sites, though only one is used in this study) was utilized to explore deuterium retention and trapping within single-crystal tungsten samples irradiated in the [High Flux Isotope Reactor (HFIR)](https://neutrons.ornl.gov/hfir) facilit ...
```
```suggestion
This case reproduces, in updated form, the analysis published in [!cite](Shimada2018). 
In the original study, a modified form of the TMAP4 code (updated to include multiple trapping sites, though only one is used in this study) was utilized to explore deuterium retention and trapping within neutron-irradiated single-crystal tungsten samples.
Samples were first irradiated in the [High Flux Isotope Reactor (HFIR)](https://neutrons.ornl.gov/hfir) facility at Oak Ridge National Laboratory and then exposed to a deuterium plasma within the [Tritium Plasma Experiment (TPE)](https://inl.gov/fusion-safety/star/) at Idaho National Laboratory. This was undertaken as part of the US-Japan Technological Assessment of Plasma Facing Components for DEMO Reactors (PHENIX) project [!citep](Katoh2017phenix, Shimada2017phenix).
```

simopier commented on doc/content/verification_and_validation/val-2i.md:
```diff
@@ -0,0 +1,177 @@
+!style halign=left
+This case reproduces, in updated form, the analysis published in [!cite](Shimada2018), where a modified form of the TMAP4 code (updated to include multiple trapping sites, though only one is used in this study) was utilized to explore deuterium retention and trapping within single-crystal tungsten samples irradiated in the [High Flux Isotope Reactor (HFIR)](https://neutrons.ornl.gov/hfir) facilit ...
+
+In the experimental phase, six single-crystal tungsten disks were prepared from micro-tensile specimens using electrical discharge machining; the dimensions of the samples after machining were $4.0 \times 4.0 \times 0.5 \text{mm}^3$. After mechanical polishing, heat treatment was not performed to remove any remaining surface damage due to the production process (e.g., shallow cracks from machinin ...
```
fixing weird rendering.
```suggestion
In the experimental phase, six single-crystal tungsten disks were prepared from micro-tensile specimens using electrical discharge machining; the dimensions of the samples after machining were $4.0 \times 4.0 \times 0.5$ mm$^3$. After mechanical polishing, heat treatment was not performed to remove any remaining surface damage due to the production process (e.g., shallow cracks from machining and parallel striations from polishing) prior to neutron irradiation. As opposed to mirror-like laboratory conditions, these were judged to represent more realistic surfaces that might be experienced in plasma facing components in fusion devices. Experimental conditions for both the HFIR and TPE phases of the experiment are shown in [val-2i-experimental-conditions].
```

simopier commented on doc/content/verification_and_validation/val-2i.md:
```diff
@@ -0,0 +1,177 @@
+In the experimental phase, six single-crystal tungsten disks were prepared from micro-tensile specimens using electrical discharge machining; the dimensions of the samples after machining were $4.0 \times 4.0 \times 0.5 \text{mm}^3$. After mechanical polishing, heat treatment was not performed to remove any remaining surface damage due to the production process (e.g., shallow cracks from machinin ...
+
+!table id=val-2i-experimental-conditions caption=The experimental (HFIR irradiation and TPE plasma) conditions for val-2i, from [!cite](Shimada2018).
+| Specimen ID | HFIR irradiation temp. ($K$) | TPE exposure temp. ($K$) | TPE exposure flux ($\text{m}^{-2} \text{s}^{-1}$) | TPE exposure fluence ($\text{m}^{-2}$)
```
```suggestion
| Specimen ID | HFIR irradiation temp. (K) | TPE exposure temp. (K) | TPE exposure flux (m$^{-2} \cdot$ s$^{-1}$) | TPE exposure fluence (m$^{-2}$)
```
units should not be in math mode.

simopier commented on doc/content/verification_and_validation/val-2i.md:
```diff
@@ -0,0 +1,177 @@
+| W26A | 1073 | 973 | $8.4 \times 10^{21}$ | $5.0 \times 10^{25}$ |
+| W26B | 1073 | 973 | $7.5 \times 10^{21}$ | $5.0 \times 10^{25}$ |
+
+Note that the HFIR irradiation dose was calculated to approximately 0.1 dpa, and the incident ion energy in TPE was approximately 100eV (more info available in [!cite](Shimada2018)). In this model, the W53A, W55A, and W26A samples were used as the targets for comparison with the model for the thermal desorption spectroscopy (TDS) measurements.
```
```suggestion
Note that the HFIR irradiation dose was calculated to approximately 0.1 dpa, and the incident ion energy in TPE was approximately 100 eV (more info available in [!cite](Shimada2018)). In this effort, the W53A, W55A, and W26A samples were used to compare experimental thermal desorption spectroscopy (TDS) measurements from modeling predictions.
```

simopier commented on doc/content/verification_and_validation/val-2i.md:
```diff
@@ -0,0 +1,177 @@
+    image_name=val-2i_temperature_history.png
+    style=width:50%;margin-bottom:2%;margin-left:auto;margin-right:auto
+    id=val-2i_temperature_history
+    caption=Temperature history used in the TMAP8 simulation for the 673 K exposure (specimen W53A), 873 K exposure (specimen W55A), and 973 K exposure (specimen W26A). This reproduces Figure 3 from [!cite](Shimada2018).
```
Add the assumed duration of the cooldown phase. The documentation above states 2-4 hours, but no exact time is given, even if one is used here. 
Also, the temperature of the sample that was held at 973 K seems to discontinuously drop to 300 K. It might not be a drop large enough to cause issues, but such behavior can cause challenges with trapping behavior, so just be mindful of that if you encounter challenges around that time.

cticenhour replied:
It does look like that - I'll see if I can adjust the function to make it smoother. Good catch.

simopier commented on doc/content/verification_and_validation/val-2i.md:
```diff
@@ -0,0 +1,177 @@
+where $C_M$ is the concentration of mobile deuterium, $D$ is the diffusivity, $C_T$ is the concentration of trapped deuterium in the material, and $S$ is the deuterium implantation source term from the TPE exposure. The diffusivity follows an Arrhenius relationship:
+
+\begin{equation}
+D = D_{0} \exp\left(-\frac{E_{a}}{k_B T}\right)
```
```suggestion
D = D_{0} \exp\left(-\frac{E_{a}}{k_B T}\right),
```

simopier commented on doc/content/verification_and_validation/val-2i.md:
```diff
@@ -0,0 +1,177 @@
+where $D_{0}$ is the pre-exponential factor, $E_{a}$ is the activation energy, $k_B$ is the Boltzmann constant, and $T$ is temperature. The implantation profile is in the form of a normal distribution, whose one-dimensional form is given by
+
+\begin{equation}
+S(x, t) = S_s(t) \frac{(1-R_{\text{ref}})}{w_s \sqrt{2 \pi}} \exp \left[-\frac{1}{2} \left(\frac{x - d_s}{w_s}\right)^2 \right],
```
```suggestion
S(x, t) = S_s(t) \frac{1-R_{\text{ref}}}{w_s \sqrt{2 \pi}} \exp \left[-\frac{1}{2} \left(\frac{x - d_s}{w_s}\right)^2 \right],
```

simopier commented on doc/content/verification_and_validation/val-2i.md:
```diff
@@ -0,0 +1,177 @@
+The model includes a single trapping site to capture deuterium retention effects observed in the TDS spectra. The trapped concentration $C_T$ evolves according to:
+
+\begin{equation}
+\frac{\partial C_T}{\partial t} = \alpha_t \frac{C_T^{\text{empty}} C_M}{N} - \alpha_r C_T
```
```suggestion
\frac{\partial C_T}{\partial t} = \alpha_t \frac{C_T^{\text{empty}} C_M}{N} - \alpha_r C_T,
```

simopier commented on doc/content/verification_and_validation/val-2i.md:
```diff
@@ -0,0 +1,177 @@
+The trapping and release rate coefficients follow Arrhenius relationships:
+
+\begin{equation}
+\alpha_t = \alpha_{t0} \exp\left(-\frac{\epsilon_t}{k_B T}\right)
```
```suggestion
\alpha_t = \alpha_{t0} \exp\left(-\frac{\epsilon_t}{k_B T}\right),
```

simopier commented on doc/content/verification_and_validation/val-2i.md:
```diff
@@ -0,0 +1,177 @@
+\end{equation}
+
+\begin{equation}
+\alpha_r = \alpha_{r0} \exp\left(-\frac{\epsilon_r}{k_B T}\right)
```
```suggestion
\alpha_r = \alpha_{r0} \exp\left(-\frac{\epsilon_r}{k_B T}\right),
```

simopier commented on doc/content/verification_and_validation/val-2i.md:
```diff
@@ -0,0 +1,177 @@
+where $\alpha_{t0}$ and $\alpha_{r0}$ are pre-factors of trapping and release rate coefficients and $\epsilon_t$ and $\epsilon_r$ are trapping and release energies. In this model, $\alpha_{t0}$ is defined as
+
+\begin{equation}
+\alpha_{t0} = \frac{D_0}{\lambda_W^2}
```
```suggestion
\alpha_{t0} = \frac{D_0}{\lambda_W^2},
```

simopier commented on doc/content/verification_and_validation/val-2i.md:
```diff
@@ -0,0 +1,177 @@
+where $\lambda_W$ is the lattice constant for tungsten.
+
+!alert note title=Typo in [!cite](Shimada2018), Section 3
+There appears to be a typo for the definition of $\alpha_{t0}$ in [!cite](Shimada2018), where $\lambda_W$ is in the denominator instead of $\lambda_W^2$. This is inconsistent with the TMAP4 input file used in the original work, so we have corrected it in the documentation here and used the correct form for trapping coefficient in the TMAP8 input file.
```
```suggestion
There appears to be a typo for the definition of $\alpha_{t0}$ in [!cite](Shimada2018), where $\lambda_W$ is in the denominator instead of $\lambda_W^2$. This is inconsistent with the TMAP4 input file used in the original work and with the units of $\alpha_{t0}$, $D_0$, and $\lambda_W^2$. We have therefore corrected it in the documentation here and used the correct form for trapping coefficient in the TMAP8 input file.
```

simopier commented on doc/content/verification_and_validation/val-2i.md:
```diff
@@ -0,0 +1,177 @@
+The release energy is defined as
+
+\begin{equation}
+\epsilon_r = E_b + \epsilon_t
```
```suggestion
\epsilon_r = E_b + \epsilon_t,
```

simopier commented on doc/content/verification_and_validation/val-2i.md:
```diff
@@ -0,0 +1,177 @@
+C_M(x = 0, t) = 0.
+\end{equation}
+
+As mentioned previously, the reflection coefficient $R_{ref}$ was adjusted as a fitting parameter to the model to account for plasma-surface interactions. To elaborate, at high deuterium flux with low diffusivity the location concentration of deuterium within the implantation depth is high. Coupled with the very high equilibrium gas pressure, near-surface precipitation follows, as described by [! ...
```
```suggestion
As mentioned previously, the reflection coefficient $R_{ref}$ was adjusted as a fitting parameter to the model to account for plasma-surface interactions. To elaborate, at high deuterium flux with low diffusivity, the location concentration of deuterium within the implantation depth is high. Coupled with the very high equilibrium gas pressure, near-surface precipitation follows, as described by [!cite](Kolasinski2013). Interconnected gas bubbles within the tungsten gives pathways for these precipitated $D_2$ molecules to escape, leading to a smaller diffusion length for the release of deuterium from solution. Subsequently, the amount of deuterium available to diffuse further past the implantation region is reduced. Thus, a portion of the implanted deuterium is "reflected" and unavailable as a source to the diffusion model.
```

simopier commented on doc/content/verification_and_validation/val-2i.md:
```diff
@@ -0,0 +1,177 @@
+## Case and Model Parameters
+
+!style halign=left
+[val-2i_diffusion_parameters] summarizes the detail of sample and experimental conditions from Shimada et al. [!citep](Shimada2018), as well as the model parameters from TODO, TODO, and estimated from validation cases in TMAP8. Where there are different parameters for each case, these are listed in order by specimen: W53A, W55A, and W26A. [val-2i_trapping_parameters] includes the trapping paramet ...
```
```suggestion
[val-2i_diffusion_parameters] summarizes the detail of sample and experimental conditions from [!cite](Shimada2018), as well as the model parameters from TODO, TODO, and estimated from validation cases in TMAP8. Where there are different parameters for each case, these are listed in order by specimen: W53A, W55A, and W26A. [val-2i_trapping_parameters] includes the trapping parameters from Karmonik et al. [!citep](karmonik1995proton) and estimated based on existing validation cases in TMAP8.
```

simopier replied:
remember to update the TODOs.

simopier commented on doc/content/verification_and_validation/val-2i.md:
```diff
@@ -0,0 +1,177 @@
+!style halign=left
+[val-2i_diffusion_parameters] summarizes the detail of sample and experimental conditions from Shimada et al. [!citep](Shimada2018), as well as the model parameters from TODO, TODO, and estimated from validation cases in TMAP8. Where there are different parameters for each case, these are listed in order by specimen: W53A, W55A, and W26A. [val-2i_trapping_parameters] includes the trapping paramet ...
+
+!table id=val-2i_diffusion_parameters caption=Experimental set up and diffusion parameters from Shimada et al. [!citep](Shimada2018) for deuterium transport in neutron-irradiated single-crystal tungsten.
```
```suggestion
!table id=val-2i_diffusion_parameters caption=Experimental set up and diffusion parameters from Shimada et al. [!citep](Shimada2018) for deuterium transport in neutron-irradiated single-crystal tungsten. Multiple parameter values correspond to [W53A, W55A, W26A].
```

simopier commented on doc/content/verification_and_validation/val-2i.md:
```diff
@@ -0,0 +1,177 @@
+| $E_a$ | Activation energy of deuterium | 0.39 | eV | [!cite](frauenfelder1969solution) |
+| $C_{M,0}$ | Initial concentration of mobile species | 0 | at. / m$^{3}$ | [!cite](Shimada2018) |
+
+!table id=val-2i_trapping_parameters caption=Trapping parameters for deuterium transport in single-crystal tungsten used in this case.
```
```suggestion
!table id=val-2i_trapping_parameters caption=Trapping parameters for deuterium transport in single-crystal tungsten used in this case. Multiple parameter values correspond to [W53A, W55A, W26A].
```

simopier commented on doc/content/verification_and_validation/val-2i.md:
```diff
@@ -0,0 +1,177 @@
+| $\beta$ | Heating rate | 10 | K/min | [!cite](Shimada2018) |
+| $l$ | Sample thickness | 0.5 | mm | [!cite](Shimada2018) |
+| $D_0$ | Diffusivity pre-exponential factor | $4.1 \times 10^{-7} / \sqrt{2}$ | m$^2$/s | [!cite](frauenfelder1969solution) and corrected for deuterium by [!cite](Causey2002) |
+| $E_a$ | Activation energy of deuterium | 0.39 | eV | [!cite](frauenfelder1969solution) |
```
```suggestion
| $E_a$ | Activation energy of deuterium diffusion | 0.39 | eV | [!cite](frauenfelder1969solution) |
```

simopier commented on doc/content/verification_and_validation/val-2i.md:
```diff
@@ -0,0 +1,177 @@
+!alert note title=Typo in [!cite](Shimada2018), Section 3
+There appears to be a typo for the definition of $\alpha_{r0}$ in [!cite](Shimada2018), where it is stated to be $10^{-13} \text{s}^{-1}$. This is inconsistent with the TMAP4 input file used in the original work, so we have corrected it in the documentation here and used the correct form for release rate coefficient in the TMAP8 input file.
+
+!table id=val-2i_implantation_parameters caption=Implantation parameters for deuterium transport in single-crystal tungsten used in this case.
```
```suggestion
!table id=val-2i_implantation_parameters caption=Implantation parameters for deuterium transport in single-crystal tungsten used in this case. Multiple parameter values correspond to [W53A, W55A, W26A].
```

simopier commented on doc/content/verification_and_validation/val-2i.md:
```diff
@@ -0,0 +1,177 @@
+!style halign=left
+Using the model described here and in [!cite](Shimada2018), the output from TMAP8 is compared to that of TMAP4 as well as the experimental data, shown in [val-2i_comparison]. The model captures the delayed release of deuterium during the TDS heating process and produces peak shapes that are consistent with both the TMAP4 model results and the experimental data.
+
+!media comparison_val-2i.py
```
The RMSPE values for TMAP8 seem absurd, even if the curves are a bit different from TMAP4, this makes no sense to me.

cticenhour replied:
Reminder of our previous discussion about this in December here: https://github.com/idaholab/tmap8/pull/325#discussion_r2633643240

In short, we need to determine a defensible way of talking about either removing the data corresponding to experimental diagnostic "noise", or a better way of calculating the error that can safely neglect the data region where we see the most divergence that is simultaneously the least "important" to the region we're comparing within.

cticenhour replied:
I had some previous notes on how to do the latter, but need to dig those up again...

cticenhour replied:
Turns out the experimental data "diagnostic floor" was a complete red herring.....This was entirely due to a bug in my RMSPE calculation for TMAP8. I perform a shift (by 12,000 s) to align the TMAP8 desorption data with that of TMAP4 and the experiment. I neglect to do this for the RMSPE calculation! When I do, the RMSPE values look much more comparable across both TMAP4 and TMAP8:


| **Case** | **RMSPE (TMAP4)** | **RMSPE (TMAP8)** |
| -- | -- | -- |
| 673 K | 9.08% | 16.14% |
| 873 K | 28.46% | 34.84% |
| 973 K | 29.35% | 30.97% |

simopier commented on doc/content/verification_and_validation/val-2i.md:
```diff
@@ -0,0 +1,177 @@
+## Results
+
+!style halign=left
+Using the model described here and in [!cite](Shimada2018), the output from TMAP8 is compared to that of TMAP4 as well as the experimental data, shown in [val-2i_comparison]. The model captures the delayed release of deuterium during the TDS heating process and produces peak shapes that are consistent with both the TMAP4 model results and the experimental data.
```
Any idea why the results are not exactly as in TMAP4?

simopier commented on doc/content/verification_and_validation/val-2i.md:
```diff
@@ -0,0 +1,177 @@
+    image_name=val-2i_comparison.png
+    style=width:50%;margin-bottom:2%;margin-left:auto;margin-right:auto
+    id=val-2i_comparison
+    caption=Comparison of TMAP8 calculations (with trapping) with TMAP4 results and experimental data during TDS process. RMSPE values comparing TMAP4 and TMAP8 to the experimental data are shown on the plot.
```
```suggestion
    caption=Comparison of TMAP8 calculations (with trapping) with TMAP4 results and experimental data during TDS process. Root mean square percentage error (RMSPE) values comparing TMAP4 and TMAP8 to the experimental data are shown on the plot.
```

simopier commented on doc/content/verification_and_validation/val-2i.md:
```diff
@@ -0,0 +1,177 @@
+Using the model described here and in [!cite](Shimada2018), the output from TMAP8 is compared to that of TMAP4 as well as the experimental data, shown in [val-2i_comparison]. The model captures the delayed release of deuterium during the TDS heating process and produces peak shapes that are consistent with both the TMAP4 model results and the experimental data.
+
+!media comparison_val-2i.py
+    image_name=val-2i_comparison.png
```
I suggest writing the RMSPE values with the color of the corresponding curve.

simopier commented on doc/content/verification_and_validation/val-2i.md:
```diff
@@ -0,0 +1,177 @@
+
+!media comparison_val-2i.py
+    image_name=val-2i_comparison.png
+    style=width:50%;margin-bottom:2%;margin-left:auto;margin-right:auto
```
The curves are hard to see in some areas due to overlap with the experimental data. Maybe add a thin black line around the curves, using something like `path_effects` maybe.

simopier commented on doc/content/verification_and_validation/val-2i.md:
```diff
@@ -0,0 +1,177 @@
+!style halign=left
+The input file for this validation case is:
+
+- [/val-2i.i]: Simulates deuterium transport in neutron-irradiated single-crystal tungsten with
```
No need for a list if there is a single input file for now.

simopier commented on doc/content/verification_and_validation/val-2i.md:
```diff
@@ -0,0 +1,177 @@
+- [/val-2i.i]: Simulates deuterium transport in neutron-irradiated single-crystal tungsten with
+  trapping effects using the parameters and model configuration described in this text.
+
+!alert note
```
```suggestion
```
We usually don't use a note for this type of information.

simopier commented on test/tests/val-2i/val-2i_temperature_history.py:
```diff
@@ -0,0 +1,114 @@
+
+# Necessary parameters
+initial_temperature = [673, 873, 973]  # K
+hold_time = [7183, 6341, 5952]  # s
+cooling_time_constant = 1200  # s
+temperature_room = 293  # K
+temperature_min = 300  # K
+TDS_start = 12000  # s
+TDS_ramp_end = 17238  # s
+TDS_end = 19038  # s
+final_temperature = 1173  # K
+temperature_ramp_rate = 10 / 60  # K/s
```
```suggestion
```
You don't use these anymore

cticenhour replied:
Good catch!

simopier commented on test/tests/val-2i/val-2i.i:
```diff
@@ -0,0 +1,323 @@
+trapping_site_fraction_1 = 0.002 # (-)
+trapping_rate_prefactor = '${fparse ${diffusivity_coefficient} / ${W_lattice_constant}^2}' # (1/s)
+release_rate_prefactor = '${units 1e13 1/s}'
+trap_per_free_1 = 1e6 # (-)
```
The trap per free coefficient is not in the equations in the documentation. You should add it and provide its value.

simopier commented on test/tests/val-2i/val-2i.i:
```diff
@@ -0,0 +1,323 @@
+[]
+
+[Variables]
+  [concentration]
```
I suggest changing this variable name to `mobile` since concentration could apply to the trapping too.

simopier commented on test/tests/val-2i/val-2i.i:
```diff
@@ -0,0 +1,323 @@
+  [trapped_1]
+    order = FIRST
+    family = LAGRANGE
+    outputs = none
```
why not? I suggest you output it. 

```suggestion
```

simopier reviewed (changes requested):
I know you are still working on this, but here is a review with comments for you to consider. 
Thank you for adding this case!

The light test currently fails:
```
DIFF test:val-2i.val-2i_673K_light FAILED (CSVDIFF)
```

edbworth commented:
> @cticenhour, do you need any help with this?

This is quite close to my current draft PR. If you would like me to help with this, I can after I finish up that work.

cticenhour commented:
I plan on making good progress on this over this week, so I think I'll be good. Thanks!

cticenhour committed:
Apply suggestions from code review

Co-authored-by: Pierre-Clement Simon <pierreclement.simon@gmail.com>

cticenhour committed:
Fixup bug in RMSPE calculation; TMAP8 time series data was not being consistently shifted

Add a note about why we do this to the comparison script

cticenhour committed:
Address more review comments

- Document use of the `trap_per_free` scaling factor in val-2i.md and
  report its value used in val-2i.i.
- Also in val-2i.md, replace TODO citation placeholders with the actual
  references and consolidate listing in input files section to being
  paragraph-based.
- Simplify variable/parameter naming in val-2i.i since only one
  trapping site is modeled. That is, drop the "_1" suffixes (trapped_1
  -> trapped, trap_per_free_1 -> trap_per_free, trapping_site_fraction_1
  -> trapping_site_fraction, etc.). Also, update the matching `cli_args`
  in the tests file for the renamed trapping_site_fraction parameter.
- Make the mobile species variable naming more clear (concentration -> mobile)
  and enable output of the trapped variable.
- Improve comparison_val-2i.py plot readability: outline the TMAP8 and TMAP4
  curves with a black stroke path effect, and color-match each RMSPE
  annotation to its corresponding curve.
